src/library_webgpu.js

```diff
@@ -1774,9 +1780,34 @@ var LibraryWebGPU = {
 
   // wgpuShaderModule
 
+  wgpuShaderModuleGetCompilationInfo__deps: ['$callUserCallback', '$stringToUTF8OnStack'],
   wgpuShaderModuleGetCompilationInfo: (shaderModuleId, callback, userdata) => {
     var shaderModule = WebGPU.mgrShaderModule.get(shaderModuleId);
-    abort('TODO: wgpuShaderModuleGetCompilationInfo unimplemented');
+    {{{ runtimeKeepalivePush() }}}
+    shaderModule["getCompilationInfo"]().then((compilationInfo) => {
+      {{{ runtimeKeepalivePop() }}}
+      callUserCallback(() => {
+        withStackSave(() => {
+          var compilationMessagesPtr = stackAlloc({{{ C_STRUCTS.WGPUCompilationMessage.__size__ }}} * compilationInfo.messages.length);
+          for (var i = 0; i < compilationInfo.messages.length; ++i) {
+            var compilationMessage = compilationInfo.messages[i];
+            var compilationMessagePtr = compilationMessagesPtr + {{{ C_STRUCTS.WGPUCompilationMessage.__size__ }}} * i;
```

Comment:
Need to initialize all the fields of WGPUCompilationMessage.
Also nit: put `message` at the top so it's in the same order as the declaration of the struct.
```suggestion
            var compilationMessagePtr = compilationMessagesPtr + {{{ C_STRUCTS.WGPUCompilationMessage.__size__ }}} * i;
            {{{ makeSetValue('compilationMessagePtr', C_STRUCTS.WGPUCompilationMessage.nextInChain, '0', '*') }}};
            var messagePtr = stringToUTF8OnStack(compilationMessage.message);
            {{{ makeSetValue('compilationMessagePtr', C_STRUCTS.WGPUCompilationMessage.message, 'messagePtr', '*') }}};
```